identity_resolver/src/resolution/resolver.rs

```diff
@@ -95,6 +98,45 @@ where
 
     delegate.apply(did.as_str()).await
   }
+
+  /// Concurrently fetches the DID Documents of the multiple given DIDs.
+  ///
+  /// # Errors
+  /// * If the resolver has not been configured to handle the method of any of the given DIDs.
+  /// * If the resolution process of any DID fails.
+  ///
+  /// ## Note
+  /// * The order of the documents in the returned vector matches that in `dids`.
+  /// * If `dids` contains duplicates, these will be resolved only once and the resolved document
+  /// is copied into the returned vector to match the order of `dids`.
+  pub async fn resolve_multiple<D: DID>(&self, dids: &[D]) -> Result<Vec<DOC>>
+  where
+    DOC: Clone,
+  {
+    let futures = FuturesUnordered::new();
+    // Create set to remove duplicates to avoid unnecessary resolution.
+    let dids_set: HashSet<D> = dids.iter().cloned().collect();
+    for did in dids_set {
+      futures.push(async move {
+        let doc = self.resolve(&did).await;
+        doc.map(|doc| (did, doc))
+      });
+    }
+    let documents: HashMap<D, DOC> = futures.try_collect().await?;
+    let mut ordered_documents: Vec<DOC> = Vec::with_capacity(dids.len());
+    // Reconstructs the order of `dids`.
+    for did in dids {
+      let doc: DOC = documents
+        .get(did)
+        .cloned()
+        // Error instead of `unwrap`.
+        .ok_or(Error::new(ErrorCause::UnspecificError {
+          message: "Internal Error".to_owned(),
+        }))?;
+      ordered_documents.push(doc);
+    }
```

Comment:
Why do we need to return it in order? The Credential validator has no assumption on the order. It will iterate the returned Vec and `find` the corresponding document. We can just use FuturesUnordered and return the vec, without sorting it imo. That also means we don't have to clone the document which is avoids the copying.
If you agree, remove the `UnspecificError` variant.

Reply:
The method will be used most of the time for validation and there we don't need the order. We can state in the documentation what the assumption is, this is a common thing.
If it's an expected thing in JS, then we should prepare that sorted return type in the bindings. I'd suggest to return a map, mapping DID to document, so we avoid the duplicates in the vector. I would really like to avoid unnecessary copies of kilobyte-large documents.

Reply:
I updated the rust to return a hashmap and did the sorting in the bindings.